examples/distributed-spmv-scaling/distributed-spmv-scaling.cpp

```diff
@@ -0,0 +1,270 @@
+    const auto start = rows_per_rank * comm->rank();
+    const auto end = gko::min(rows_per_rank * (comm->rank() + 1), mat_size);
+
+    auto A_data =
+        gko::matrix_data<ValueType, IndexType>(gko::dim<2>{mat_size, mat_size});
+
+    for (IndexType row = start; row < end; row++) {
+        auto i = row / dp;
+        auto j = row % dp;
+        for (IndexType d_i = -1; d_i <= 1; d_i++) {
+            for (IndexType d_j = -1; d_j <= 1; d_j++) {
+                if (!restricted || (d_i == 0 || d_j == 0)) {
+                    auto col = j + d_j + (i + d_i) * dp;
+                    if (col >= 0 && col < mat_size) {
+                        A_data.nonzeros.emplace_back(row, col,
+                                                     gko::one<ValueType>());
+                    }
+                }
+            }
+        }
+    }
+
+    return A_data;
+}
+
+
+/**
+ * Generates matrix data for a 3D stencil matrix. If restricted is set to true,
+ * creates a 7-pt stencil, if it is false creates a 27-pt stencil. If
+ * strong_scaling is set to true, creates the same problemsize independent of
+ * the number of ranks, if it false the problem size grows with the number of
+ * ranks.
+ */
+template <typename ValueType, typename IndexType>
+gko::matrix_data<ValueType, IndexType> generate_3d_stencil(
+    const IndexType dp, std::shared_ptr<gko::mpi::communicator> comm,
+    bool restricted, bool strong_scaling)
+{
+    const auto mat_size =
+        strong_scaling ? dp * dp * dp : dp * dp * dp * comm->size();
+    const auto rows_per_rank = gko::ceildiv(mat_size, comm->size());
+    const auto start = rows_per_rank * comm->rank();
+    const auto end = gko::min(rows_per_rank * (comm->rank() + 1), mat_size);
+
+    auto A_data =
+        gko::matrix_data<ValueType, IndexType>(gko::dim<2>{mat_size, mat_size});
+
+    for (IndexType row = start; row < end; row++) {
+        auto i = row / (dp * dp);
+        auto j = (row % (dp * dp)) / dp;
+        auto k = row % dp;
+        for (IndexType d_i = -1; d_i <= 1; d_i++) {
+            for (IndexType d_j = -1; d_j <= 1; d_j++) {
+                for (IndexType d_k = -1; d_k <= 1; d_k++) {
+                    if (!restricted ||
+                        ((d_i == 0 && d_j == 0) || (d_i == 0 && d_k == 0) ||
+                         (d_j == 0 && d_k == 0))) {
+                        auto col =
+                            k + d_k + (j + d_j) * dp + (i + d_i) * dp * dp;
+                        if (col >= 0 && col < mat_size) {
+                            A_data.nonzeros.emplace_back(row, col,
+                                                         gko::one<ValueType>());
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    return A_data;
+}
+
+
+int main(int argc, char* argv[])
+{
+    const auto fin = gko::mpi::init_finalize(argc, argv);
+    // Use some shortcuts. In Ginkgo, vectors are seen as a gko::matrix::Dense
```

Comment:
This comment, starting at "In Gingko ..." is a bit convoluted, may be leave it out all together?